aws-rds-cfn-common/src/main/java/software/amazon/rds/common/handler/Tagging.java

```diff
@@ -184,56 +185,52 @@ private static RemoveTagsFromResourceRequest removeTagsFromResourceRequest(
                 .build();
     }
 
-    public static ErrorRuleSet bestEffortErrorRuleSet(
+    public static ErrorRuleSet getUpdateTagsAccessDeniedRuleSet(
             final TagSet tagsToAdd,
             final TagSet tagsToRemove
     ) {
-        return bestEffortErrorRuleSet(tagsToAdd, tagsToRemove, SOFT_FAIL_TAG_ERROR_RULE_SET, HARD_FAIL_TAG_ERROR_RULE_SET);
+        return getUpdateTagsAccessDeniedRuleSet(tagsToAdd, tagsToRemove, STACK_TAGS_ERROR_RULE_SET, RESOURCE_TAG_ERROR_RULE_SET);
     }
 
-    public static ErrorRuleSet bestEffortErrorRuleSet(
+    public static ErrorRuleSet getUpdateTagsAccessDeniedRuleSet(
             final TagSet tagsToAdd,
             final TagSet tagsToRemove,
-            final ErrorRuleSet softFailErrorRuleSet,
-            final ErrorRuleSet hardFailErrorRuleSet
+            final ErrorRuleSet stackTagsErrorRuleSet,
+            final ErrorRuleSet resourceTagsErrorRuleSet
     ) {
-        // Only soft fail if the customer provided no resource-level tags
+        /* If the tagging operation comes across an AccessDenied error, we will throw an UnauthorizedTaggingOperation
+        errorCode for stack level tags. For Resource tags, if they are included, we will throw an AccessDenied error.
+        This is done to ensure backward compatibility. */
         if (tagsToAdd.getResourceTags().isEmpty() && tagsToRemove.getResourceTags().isEmpty()) {
-            return softFailErrorRuleSet;
+            return stackTagsErrorRuleSet;
         }
-        return hardFailErrorRuleSet;
+        return resourceTagsErrorRuleSet;
     }
 
-    public static <M, C extends TaggingContext.Provider> ProgressEvent<M, C> safeCreate(
+    public static <M, C extends TaggingContext.Provider> ProgressEvent<M, C> createWithUnauthorizedTagging(
```

Comment:
nit: the name of this method implies that we are creating the resource while unauthorized. We are performing creation of the resource while also using tagging mechanism fallback. `createWithTaggingFallback`?

Reply:
Done